pkg/controller/jobs/pod/pod_controller.go

```diff
@@ -897,9 +897,10 @@ func (p *Pod) ConstructComposableWorkload(ctx context.Context, c client.Client,
 
 	wl := &kueue.Workload{
 		ObjectMeta: metav1.ObjectMeta{
-			Namespace:  p.pod.GetNamespace(),
-			Labels:     map[string]string{},
-			Finalizers: []string{kueue.ResourceInUseFinalizerName},
+			Namespace:   p.pod.GetNamespace(),
+			Labels:      map[string]string{},
+			Finalizers:  []string{kueue.ResourceInUseFinalizerName},
+			Annotations: p.pod.Annotations,
```

Comment:
So that I understand this better: who is responsible for setting annotations on the pod (or job)? I suppose that will be as part of the pod/job config that users submit to kueue?

Reply:
Yes, exactly